### DISCUSSION TITLE
SOFA crashes when Hyperelasticity is applied

### DISCUSSION THREAD

**Discussion by andersonnardin** [SofaDefrost/SoftRobots#224] [Help - Building / Configuring] [answered]
Dear colleagues,
I tried to use the hyperelasticity force field component, but it caused crashes on SOFA.
For example, the issue can be replicated in the pneunet tutorial step 7. If you add:
`finger.addObject('TetrahedronHyperelasticityFEMForceField', name="FEM", ParameterSet="5000 7000 10", materialName="MooneyRivlin")`.  I can load the scene without error but cannot animate it.

So, what is the error here? I tried different scenes, different parameters, and different constitutive laws. The result is always the same. Is there a way to simulate a hyperelastic behavior of the material for this scene, for example?

In advance, thank you for your help.

**Comment by alxbilger** [comment, marked as answer]
Thanks for reporting the problem. I can confirm it as I am able to reproduce it.
The problem comes from the topology of the object, in particular it's probably due to the loaded mesh itself. We already encountered problems with meshes exported from Gmsh (https://github.com/sofa-framework/sofa/pull/2607, https://github.com/sofa-framework/sofa/pull/2638). The trick is to skip the edge list provided by the mesh file.

Instead of

```python
finger.addObject('MeshTopology', src='@loader', name='container')
```

we should do

```python
finger.addObject('MeshTopology', tetrahedra='@loader.tetrahedra', position='@loader.position', name='container')
```

If I do that, the scene does not crash.

Notes for the developers:

- We should print an explicit error message in this case
- We must find all occurrences of this problem in the examples
